### ISSUE TITLE
Improve Modules __call__ and method arguments on RTD

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#1049]
(from @bastings):

The docstring of [Dense](https://github.com/google/flax/blob/master/flax/linen/linear.py#L140) is pretty good but it doesn't not show up nicely on [RTD](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Dense.html). The `__call__` method seems to be missing.

Also the arguments to some modules are represented badly, e.g.: https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Conv.html

**Comment by cccntu** [comment]
> The `__call__` method seems to be missing.

 I just noticed the same issue when doing #1090. Look like every linen module has the same issue.
 
This seems to be the reason: https://stackoverflow.com/a/62161540
Hope that helps.

**Comment by marcvanzee** [comment]
Interesting, thanks for noting this @cccntu! Indeed there seems to be a problem with all Modules. I've changed the title and changed the priority from P2 -> P1.

**Comment by marcvanzee** [comment]
Another issue with RTD from @andsteing 

"I was just checking Flax docs for something... Has it always been like this?
https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Conv.html
I find that super confusing to navigate."

I agree this looks terrible, so I'm extending the scope of this issue to also resolve that.

**Comment by marcvanzee** [comment]
This is finally resolved after #1416! (FYI @bastings)

We added some planned follow-up work in #1676 .
